RIOT-OS/RIOT
# pkg/etl: Add the embedded template library (etl)

### Contribution description

This PR contains the [ETL](https://www.etlcpp.com/) as package. It offers a similar API to the C++ stdlib without using the heap. Additionally it reimplements some parts of the stdlib like type-traits if there is no stdlib available.

I've used the ETL in embedded projects with RIOT for the containers and liked it.
Since I am currently implementing some C++ additions to RIOT I noticed that the AVR platform has no stdlib support (and therefore no type traits). Type traits are really convenient even without stdlib support. When using the etl package it's possible to take advantage of type traits during compile time without increasing ram or rom size.

### Testing procedure
The ETL is already heavily unit tested with several compilers. [[1](https://www.etlcpp.com/home.html)][[2](https://github.com/ETLCPP/etl/tree/master/test)]

I've just added one little test using some type traits, a vector and a string. This internally uses multiple parts of the ETL to verify that compilation works with multiple platforms. The test compares that the printed string is valid.
